django/core/management/commands/dumpdata.py

```diff
@@ -184,7 +198,36 @@ def get_objects(count_only=False):
             if output and self.stdout.isatty() and options['verbosity'] > 0:
                 progress_output = self.stdout
                 object_count = sum(get_objects(count_only=True))
-            stream = open(output, 'w') if output else None
+            if output:
+                file_root, file_ext = os.path.splitext(output)
+                compression_formats = {
+                    '.gz': (gzip.open, {}, output),
+                    '.bz2': (open, {}, file_root),
+                    '.lzma': (open, {}, file_root),
+                    '.xz': (open, {}, file_root),
```

Comment:
don't put `bz2`, `lzma` or `xz` in here with dummy values, since the fallback `except KeyError` will already use the same fallbacks?

Reply:
Actually it' not a dummy value, here I followed the Nick's suggestion, later confirmed by [Marius](https://github.com/django/django/pull/13797#pullrequestreview-556905890), in case the compression format is not supported:
> Strip the extension and continue with an uncompressed file, but show a warning.

The except instead is needed for other file extension with no compression (e.g. .txt , .json, ...)

Reply:
Ah I see, okay then.